Backend/Backend/Services/EpisodeService.cs

```diff
@@ -54,11 +84,28 @@ public class EpisodeService : IEpisodeService
 
 
         public async Task<bool> DeleteEpisode(Episode episode, DeleteEpisodeRequest deleteEpisodeRequest)
-        {  
-           await _fileService.DeleteFile(episode.AudioFileId.ToString()!);
-           _db.Remove(episode);
-           await _db.SaveChangesAsync();
-           return true;
+        {
+            Guid? guid = Guid.Parse(deleteEpisodeRequest.EpisodeId);
+            Files? file = await _db.File!.FirstOrDefaultAsync(u => u.FileId == episode.AudioFileId);
+
+            if (file != null)
+            {
+               var deleted = _fileService.DeleteFile(file.Path!);
+               if(deleted == true)
+                {
+                _db.Remove(file);
+
+                }
+               else { throw new Exception("Failed to Delete File"); }
+                _db.Remove(episode);
+                await _db.SaveChangesAsync();
+                return true;
+            }
+            else
+            {
+                throw new Exception("File Not Found");
```

Comment:
File not found exception will already happen if we fail the  _fileService.DeleteFile, since it tries to find if the path exists. We could just output the exception there

Reply:
Never mind, That was my mistake. This is fine